### DISCUSSION TITLE
Before run replication, need to set read-only mode to harbor?

### DISCUSSION THREAD

**Discussion by zhucan** [goharbor/harbor#19760] [Q&A] [answered]
When run replication,  if we push the images to the src repo(not set read-only mode), the new image wll be replica from src to des?

**Comment by zyyw** [comment, marked as answer]
You can set `Replication mode` to `Push-based` and set `Trigger Mode` to `Event Based`, then whenever you push an artifact to the source registry, it should be replicated to the destination registry based on the replication rule.

**Reply by zhucan** [reply]
if we set Trigger Mode to Event Based, when push image to the source registry, only replication the new image not all image?

**Reply by zyyw** [reply]
Yes.

**Reply by zhucan** [reply]
so, my question is, if we set Trigger Mode to Manually, it will not replica  the new image?

**Reply by zyyw** [reply]
If the `Trigger Mode` of the replication rule is set to `Manually`, no images will be replicated unless you manually trigger the replication rule.

**Comment by microyahoo** [comment]
hi @zyyw, If I understand correctly, you mean that two replication rules need to be set for online replication, then manualy trigger the `Manual` replication rule, and `Event Based` is triggered automatically. These two rules running at the same time.
1. One is to set `Replication mode` to `Push-based` and set `Trigger Mode` to `Event Based`, which is incremental replication.
2. One is to set `Replication mode` to `Push-based` and set `Trigger Mode` to `Manual`, which is full replication.

**Comment by microyahoo** [comment]
hi @zyyw, @chlins, how to sync users to another harbor? Thanks.
